Covid19Radar/Covid19Radar/Repository/ServerConfigurationRepository.cs

```diff
@@ -0,0 +1,301 @@
+﻿// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Covid19Radar.Common;
+using Newtonsoft.Json;
+using Xamarin.Essentials;
+
+namespace Covid19Radar.Repository
+{
+    public interface IServerConfigurationRepository
+    {
+        public string[] Regions { get; set; }
+
+        public string UserRegisterApiEndpoint { get; set; }
+
+        public string DiagnosisKeyRegisterApiEndpoint { get; set; }
+
+        public virtual IList<string> DiagnosisKeyRegisterApiUrls
+        {
+            get
+            {
+                return Regions
+                    .Select(region => DiagnosisKeyRegisterApiEndpoint.Replace(ServerConfiguration.PLACEHOLDER_REGION, region))
+                    .Where(url => url != null)
+                    .Distinct()
+                    .ToList();
+            }
+        }
+
+        public string DiagnosisKeyListProvideServerEndpoint { get; set; }
+
+        public virtual string GetDiagnosisKeyListProvideServerUrl(string region)
+            => DiagnosisKeyListProvideServerEndpoint.Replace(ServerConfiguration.PLACEHOLDER_REGION, region);
+
+        public virtual IList<string> DiagnosisKeyListProvideServerUrls
+        {
+            get
+            {
+                return Regions
+                    .Select(region => DiagnosisKeyListProvideServerEndpoint.Replace(ServerConfiguration.PLACEHOLDER_REGION, region))
+                    .Where(url => url != null)
+                    .Distinct()
+                    .ToList();
+            }
+        }
```

Comment:
[修正任意]
アロー演算子を使うとスッキリ書けます。

```suggestion
        public virtual IList<string> DiagnosisKeyListProvideServerUrls => Regions
            .Select(region => DiagnosisKeyListProvideServerEndpoint.Replace(ServerConfiguration.PLACEHOLDER_REGION, region))
            .Where(url => url != null)
            .Distinct()
            .ToList();
```

Reply:
eb81000 で対応しました。